File: backend/backend.Domain/src/Entities/Product.cs
namespace backend.Domain.src.Entities;

public class Product : BaseEntity
{
    public string Title { get; set; } = String.Empty;
    public string Description { get; set; } = "";
    public decimal Price { get; set; }
    public int Inventory { get; set; }
    public List<string> Images { get; set; }
    public Guid CategoryId { get; set; }
    public Category Category { get; set; }
    public List<Review> Reviews { get; set; }
    public List<OrderItem> OrderItems { get; set; }
}

File: frontend/src/types/Product.ts
import { Category } from "./Category";
export interface Product {
  id: string;
  title: string;
  description: string;
  price: number;
  PictureUrl: string;
  category: Category;
  reviews: Review[];
}

export interface Review {
  id: string;
  comment: string;
  rating: number;
  productId: string;
  userId: string;
}

interface ProductDetails {
  id: string;
  title: string;
  description: string;
  price: number;
  image: string;
  category: Category;
  reviews: [
    {
      id: string;
      comment: string;
      rating: number;
      productId: string;
      userId: string;
    }
  ];
}
export default Product;
